SimplifiedTests/NYPLFinderBusinessLogicTests.swift

```diff
@@ -0,0 +1,82 @@
+//
+//  NYPLFinderBusinessLogicTests.swift
+//  SimplyETests
+//
+//  Created by Ernest Fan on 2021-05-26.
+//  Copyright © 2021 NYPL Labs. All rights reserved.
+//
+
+import XCTest
+import CoreLocation
+
+@testable import SimplyE
+
+class NYPLFinderBusinessLogicTests: XCTestCase {
+
+  var businessLogic: NYPLLibraryFinderBusinessLogic!
+  var libraryRegistryMock: NYPLLibraryRegistryFeedRequestHandlerMock!
+  var libraryAccounts: [Account]!
+  
+  override func setUpWithError() throws {
+    libraryAccounts = [Account]()
+    let testFeedUrl = Bundle.init(for: OPDS2LibraryRegistryFeedTests.self)
+      .url(forResource: "OPDS2LibraryRegistryFeed", withExtension: "json")!
+    let data = try Data(contentsOf: testFeedUrl)
+    let feed = try OPDS2LibraryRegistryFeed.fromData(data)
+    for catalog in feed.catalogs {
+      libraryAccounts.append(Account(libraryCatalog: catalog))
+    }
+    
+    let userAccounts: [Account] = [libraryAccounts.first!]
+    
+    libraryRegistryMock = NYPLLibraryRegistryFeedRequestHandlerMock()
+    
+    businessLogic = NYPLLibraryFinderBusinessLogic(userAccounts: userAccounts, libraryRegistry: libraryRegistryMock)
+  }
+
+  override func tearDownWithError() throws {
+    libraryRegistryMock.requestUrl = nil
+    libraryRegistryMock.libraryAccounts = [Account]()
+    
+    businessLogic.userLocation = nil
+    businessLogic.newLibraryAccounts = [Account]()
+  }
+
+  func testLibraryRegistryRequestUrlWithSearchKeyword() throws {
+    businessLogic.requestLibraryList(searchKeyword: "test") { (success) in
+      XCTAssertEqual(self.libraryRegistryMock.requestUrl?.absoluteString, "http://librarysimplified.org/terms/rel/search?query=test&stage=all")
```

Comment:
we should add an expectation that the closure is called. Otherwise the test might still pass if we don’t call the completion.